skbeam/core/image.py

```diff
@@ -170,3 +171,64 @@ def construct_circ_avg_image(radii, intensities, dims=None, center=None,
     CIMG = np.interp(radial_val, radii, intensities, right=0)
 
     return CIMG
+
+
+def construct_rphi_avg_image(radii, angles, image, mask=None,
+                             center=None, dims=None, pixel_size=(1, 1)):
+    '''  Construct a 2D image in rectangular coordinates from
+        a polar coordinate image. Assumes a 2D array of data. If data is
+        missing, use mask.
+
+        radii : 1d array of coordinates, ascending order. The radii
+            must be in units of pixels.
+
+        angles : 1d array of coordinates, ascending order
+            must be in units of radians
+            domain *must* be 0 to 2*pi monotonically increasing (or a subset)
+            if it is not, code will *try* to repair this
+
+        image : 2d array the image to interpolate from
+
+        mask : 2d array, optional
+            the masked data (0 masked, 1 not masked). Defaults to None,
+            which means assume all points are valid.
+
+        center : 2 tuple of floats, optional
+
+        dims : the new image dimensions, in terms of pixel values
+            (same units as radii)
+
+    '''
+    if dims is None:
+        if center is not None:
+            raise ValueError("Specifying a dims but not a center does not "
+                             "make sense and may lead to unexpected results.")
+        # round up, also take into account pixel size change
+        maxr_y, maxr_x = (int(np.max(radii/pixel_size[0])+.5),
+                          int(np.max(radii/pixel_size[1])+.5))
+        dims = 2*maxr_y+1, 2*maxr_x+1
+
+    if center is None:
+        center = (dims[0]-1)/2., (dims[1] - 1)/2.
+
+    if mask is None:
```

Comment:
good idea. For some reason, doing this with RectBivariateSpline leads to all np.nans (even by setting degree of interpolation to kx=ky=1, not 3). I would have assumed that linear interpolation would just averaged neighboring pixels (or maybe a few negihboring pixels but not more).
I have changed this to use a simpler version of linear interpolation. I couldn't find it in any library, so I modified some code from stackoverflow. I retrofitted it to work with dimensions that wrap around. I am not sure if this is the best approach yet. Will write tests later.

If you had comments I would like to hear them, in case I am taking a longwinded approach... thanks for the review